### PR TITLE
winpr/thread: small cleanups

### PR DESCRIPTION
**Pull request by bmiklautz**
Also adds error checking of winpr_StartThread in winpr_StartThread.

### CODE REVIEW

**Comment by bmiklautz** [comment]
@freerdp-bot test!

**Comment by bmiklautz** [comment]
Hmm looks like the build farm has network troubles. Retrying later.

**Comment by bmiklautz** [comment]
@freerdp-bot test

File: winpr/libwinpr/thread/thread.c [hunk 1]
```diff
@@ -331,7 +328,6 @@ static void* thread_launcher(void* arg)
 			cleanup_handle(thread);
 	}
 	pthread_exit((void*) (size_t) res);
```
Comment (mfleisz) [inline winpr/libwinpr/thread/thread.c]: Shouldn't we just return res here? Not returning anything might cause compiler warnings and simply returning res should result in an implicit pthread_exit(res); call according to docs.

Comment (bmiklautz) [inline winpr/libwinpr/thread/thread.c]: @mfleisz excellent point. I'll update the PR. Thank you.

**Comment by mfleisz** [comment]
+1

**Comment by akallabeth** [comment]
+1

**Pull request merged**